webapp/src/main/resources/application.yml

```diff
@@ -30,13 +30,19 @@ server:
 
 spring:
   datasource:
-    url: jdbc:mysql://localhost:3306/warehouse?useSSL=false&useLegacyDatetimeCode=false&characterEncoding=utf8
+    url: "\
+      jdbc:mysql://${spring.datasource.url-server:localhost:3306}/${spring.datasource.url-schema:warehouse}\
+      ?useSSL=${spring.datasource.use-ssl:false}\
+      &useLegacyDatetimeCode=${spring.datasource.use-legacy-datetime-code:false}\
+      &characterEncoding=${spring.datasource.character-encoding:utf8}\
```

Comment:
Where do these get set?

Reply:
Mostly they never get set since the defaults are good. But the individual params can be set in the configuration yml. An example is the test profile below ...